tests/unit/test_record.py

```diff
@@ -216,3 +227,60 @@ def test_func(a: int, b: str, c: Optional[str] = None) -> str:
     assert recorder._records_by_type["TestAutoRecord"][-1].params.a == 123
     assert recorder._records_by_type["TestAutoRecord"][-1].params.b == "abc"
     assert recorder._records_by_type["TestAutoRecord"][-1].result.return_val == "123abc"
+
+
+def test_recorded_function_with_override() -> None:
+    os.environ["DBT_RECORDER_MODE"] = "Record"
+    recorder = Recorder(RecorderMode.RECORD, None)
+    set_invocation_context({})
+    get_invocation_context().recorder = recorder
+
+    @supports_replay
+    class Recordable:
+        @auto_record_function("TestAuto")
+        def test_func(self, a: int) -> int:
+            return 2 * a
+
+    class RecordableSubclass(Recordable):
```

Comment:
I am assuming we have more than 2 layers(baseAdapter, sqlAdapter, SnowflakeAdapter), does it still work?(I think so but a bit harder to reason through, maybe consider expand this test a bit?

Reply:
That's a good point, I also need to inject a modification to the subclasses own __init_subclass__ function. I'll add that in the next revision.